### ISSUE TITLE
Block editor opens in default size - possible to autosize the block editor window? 

### ISSUE THREAD

**Issue by olavand** [jmoenig/Snap--Build-Your-Own-Blocks#188]
Autosize relative to scriptlength, so that all code is shown without manually changing it.

**Comment by studej** [comment]
<b>code:</b>
in `byob.js` under line `1696` (`this.addButton('cancel', 'Cancel');`) insert this:

```
//find recommended width and height of custom block definition script
var width = 0, height = 0,
	helper = function(currBlock, depth){ //depth is used to compute additional padding of nested scripts in C-shaped slots
		do{
			currBlock.inputs().forEach(function(input){	//iterate through current block inputs and do recursive call for every CSlotMorph
				if(input instanceof CSlotMorph && input.nestedBlock() != null){	
					helper(input.nestedBlock(), depth + 1);
				}
			});
			var temp = currBlock.width() + 7 * depth; //there should be cSlotPadding from blocks.js, but it is not accessible. cSlotPadding value is 4, but it turned out from my tests, that better value is 7 (actually used in code). And maybe there should be 7 * scale of blocks, but it is not accessible too.
			if(temp > width){ //if current block width (temp) is greater than the last greatest one
				width = temp;
			}
			if(depth == 0){ //instead of adding height of every block it is better to take care only of first level blocks (0 depth). Height of block with C-shaped slot is already computed based on nested script.
				height += currBlock.height();
			}			
		}while((currBlock = currBlock.nextBlock()) != null);
	};

helper(proto, 0); //start from prototype hat block, depth 0
var additionalPadding = 50;
width += this.padding * 2 + scripts.cleanUpMargin * 2 + additionalPadding;
height += this.height() - scriptsFrame.height() + additionalPadding / 2;
var MIN_W = 375, MIN_H = 300,
	MAX_W = 750, MAX_H = 600,
	final_w = ((width < MIN_W) ? MIN_W : ((width > MAX_W) ? MAX_W : width)),
	final_h = ((height < MIN_H) ? MIN_H : ((height > MAX_H) ? MAX_H : height));
this.setExtent(new Point(final_w, final_h)); //set final bounds
```

<b>current issues:</b>
- for `proto` is called `forEach` once in `helper` function
- `cSlotPadding` and `scale` are not static (accessible) in current design (correct me if I am wrong)
- height is not properly computed if script in Snap contains block with C-shaped slot (screenshot)

![custom_block_editor_size2](https://cloud.githubusercontent.com/assets/7092949/5607378/a01fc290-9459-11e4-9177-b9641791601f.png)

**Comment by cycomachead** [comment]
Fixed with 6061403a89bbbdd448a9086d3f443b5a5f88a330

@brianharvey @jmoenig This could be closed. :)

**Comment by jmoenig** [comment]
Yep, closing. Thanks!
